wiki/Publishing-guidelines/Writing-functional-specs.md

```diff
@@ -0,0 +1,383 @@
+
+> ![image](https://user-images.githubusercontent.com/3793636/117873919-f6faba80-b265-11eb-81a5-039bdcd822e8.png)  here is a link to the functional specs template: [[fs template]]
+
+<br />
+
+### Meta data
+
+The meta data is used for organizational purposes to help the reader find what they are looking for while also providing the team with language and breakdown of how a component is formed. Some of these items are auto generated, so there is no effort on your part to update while others are manual. 
+
+#### Table of contents
+The table of contents in the template is auto generated. As long as you are using proper markdown headings it should find them just fine, and list the page outline up at the top of the page. So make sure to include the table of contents section template in your component wiki page.
+
+#### Dependencies
+The dependency list is one of the most important aspects of the functional specs. It communicates to the developer, and creates almost this written agreement between design and development what components should be used when building a component out. 
+
+###### How to add a dependency
+
+1. Find the dependency list in your component's template.
+2. Add a new list item
+3. Link to the component using the Wiki linking strategy.
+
+<details>
+  <summary><strong>Example</strong></summary><br />
+
+```html
+- [[Component or wiki page name 1]]
+- [[Component or wiki page name 2]]
+- [[Component or wiki page name 3]]
+```
+
+> ![image](https://user-images.githubusercontent.com/3793636/117873919-f6faba80-b265-11eb-81a5-039bdcd822e8.png)  Notice the double brackets with the file name `[[File name]]` in-between. This is different from a markdown link `[text](url)`.
+
+<br />
+</details><br /><br />
+
+#### Used by
+
+By properly linking dependencies we will also run an automation script to collect all the dependencies and create a used by list in addition. Used by components once populated communicates to the reader where this component is used.
+
+###### How to add a used by
+
+> ![image](https://user-images.githubusercontent.com/3793636/117873919-f6faba80-b265-11eb-81a5-039bdcd822e8.png)  Keep dependencies up to date, and leverage used by template section in the component wiki page.
+
+#### Backlinks
+Every time a wiki document references a component other than itself it creates a back link. Those back links are listed out on that component's page as a snippet and categorized by pages. When clicked it will go to the section of the page where it was talked about, and highlight its contents. A backlink doesn't have to be used as a depency to work, and this list is automated. As long as you include it on the page it'll render its content if available.
```

Comment:
```suggestion
When a wiki document references another component, it creates a back link. Back links are autogenerated and are listed by page, along with a snippet of text from the page for context. The links take you directly to the mention on the page. 
```

Reply:
@photodow I wasn't sure what you meant by the dependency comment so I left it out but we can add back in.

Reply:
I can go either way. I was just trying to say that it ignores the links in the dependency list, and trying to differentiating it from some of the other sections. I think what you have here is fine though.